python/cairo-ec/src/cairo_ec/ec_ops.cairo

```diff
@@ -5,12 +5,22 @@ from starkware.cairo.lang.compiler.lib.registers import get_fp_and_pc
 from cairo_ec.curve.g1_point import G1Point
 from cairo_ec.circuits.ec_ops_compiled import ec_add as ec_add_unchecked, ec_double
 from cairo_ec.uint384 import uint384_is_neg_mod_p, uint384_eq_mod_p, felt_to_uint384
-from cairo_ec.circuits.ec_ops_compiled import assert_is_on_curve
+from cairo_ec.circuits.ec_ops_compiled import assert_is_on_curve_or_fallback
 
-// @notice Try to get the point from x.
-// @return y The y point such that (x, y) is on the curve if success is 1, otherwise (g*h, y) is on the curve
-// @return is_on_curve 1 if the point is on the curve, 0 otherwise
-// @dev g is the generator point and h is the hash of the message
+// @notice Attempts to derive a y-coordinate for a given x on an elliptic curve.
+// @return y A candidate y-coordinate; if is_on_curve = 1, (x, y) is on the curve; if 0, y is a fallback value.
+// @return is_on_curve 1 if (x, y) lies on the curve, 0 if not.
+// @dev Given x, computes y such that y^2 = x^3 + ax + b mod p when is_on_curve = 1, confirming (x, y) is on the curve.
+//      If is_on_curve = 0, y^2 = g * (x^3 + ax + b) mod p, where g is a scalar tweak (not a point),
```

Comment:
by the way: what's G? I cannot link it to a property of the curve. So I assume it's just a scalar, but how was it chosen ?

Reply:
but a generator is a 2d point? Here we have a single scalar?